src/systemctl/systemctl.c

```diff
@@ -5475,8 +5476,10 @@ static int daemon_reload(int argc, char *argv[], void *userdata) {
         /* Note we use an extra-long timeout here. This is because a reload or reexec means generators are rerun which
          * are timed out after DEFAULT_TIMEOUT_USEC. Let's use twice that time here, so that the generators can have
          * their timeout, and for everything else there's the same time budget in place. */
-
-        r = sd_bus_call(bus, m, DEFAULT_TIMEOUT_USEC * 2, &error, NULL);
+        if (arg_timeout <= DEFAULT_TIMEOUT_USEC * 2)
+                r = sd_bus_call(bus, m, DEFAULT_TIMEOUT_USEC * 2, &error, NULL);
+        else
+                r = sd_bus_call(bus, m, arg_timeout, &error, NULL);
```

Comment:
Using `MAX()` macro is simpler, I think.

Reply:
Thanks.
I'll update this PR according to your suggestions.